OpenRA.Game/Renderer.cs

```diff
@@ -84,15 +84,16 @@ public Renderer(IPlatform platform, GraphicSettings graphicSettings)
 			TempBufferSize = graphicSettings.BatchSize;
 			SheetSize = graphicSettings.SheetSize;
 
-			WorldSpriteRenderer = new SpriteRenderer(this, Context.CreateShader("combined"));
+			var combinedBindings = new CombinedShaderBindings();
+			WorldSpriteRenderer = new SpriteRenderer(this, Context.CreateShader(combinedBindings));
```

Comment:
For a future PR (definitely out of scope here): Would it be possible to have a very minimal (no palettes, no secondary sheet, no tint etc) renderer hardcoded for use by the load screen, and then defer all of these ones until after `ModData` is available? This would then let shaders modify their behaviour (e.g. changing the shader code that is loaded) based on the manifest data.

Reply:
world renderers can be deferred without errors. The instantiation of whole classes can be deferred.

The problem comes with the UI renderer, there are some issues when trying to delay its shader. Granted I haven't fully investigated all the issues

Reply:
Those problems will be due to the loadscreens right? this is why I suggested keeping a minimal one around for that.

Reply:
I've delayed the initialisation on ModelRenderer in #21066